src/comb_filter.rs

```diff
@@ -21,26 +27,62 @@ pub enum Error {
 
 impl CombFilter {
     pub fn new(filter_type: FilterType, max_delay_secs: f32, sample_rate_hz: f32, num_channels: usize) -> Self {
-        todo!("implement")
+        CombFilter {
+            filter_type: filter_type,
+            gain: 0.0,
+            max_delay_secs: max_delay_secs,
+            sample_rate_hz: sample_rate_hz,
+            max_delay_samples: (max_delay_secs * sample_rate_hz) as usize,
+            num_channels: num_channels
+        }
     }
 
     pub fn reset(&mut self) {
-        todo!("implement")
+        self.gain = 0.0;
+        self.max_delay_secs = 0.0;
     }
 
     pub fn process(&mut self, input: &[&[f32]], output: &mut [&mut [f32]]) {
-        todo!("implement");
+        let mut delay_line = vec![vec![0.0 as f32; self.num_channels]; self.max_delay_samples];
+        for i in 0..input.len() {
+            for channel in 0..self.num_channels {
+                output[i][channel] = input[i][channel] + self.gain * delay_line[self.max_delay_samples - 1][channel];
+                for delay_index in (1..self.max_delay_samples).rev() {
+                    delay_line[delay_index][channel] = delay_line[delay_index - 1][channel];
+                }
+                delay_line[0][channel] = match self.filter_type {
```

Comment:
Might want to add a try or something at the beginning of `process()` to check if the delay line is empty or `max_delay_samples` is zero.